backend/app/controllers/spree/admin/promotions_controller.rb

```diff
@@ -41,8 +41,9 @@ def collection
 
         @collection = super
         @search = @collection.ransack(params[:q])
-        @collection = @search.result(distinct: true).
+        @collection = @search.result.
```

Comment:
It seems that people have hit this `distinct` issue previously with product:

- https://github.com/solidusio/solidus/blob/master/core/app/models/spree/product/scopes.rb#L184-L207